<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LibraryManagementApp</class>
 <widget class="QMainWindow" name="LibraryManagementApp">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>814</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Library Management App</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="usesScrollButtons">
       <bool>true</bool>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>false</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="BooksTab">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <attribute name="title">
        <string>Books</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QTableWidget" name="bookTable">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::CurrentChanged</set>
          </property>
          <property name="showGrid">
           <bool>true</bool>
          </property>
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="columnCount">
           <number>13</number>
          </property>
          <column>
           <property name="text">
            <string>ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>ISBN</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Barcode</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Title</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Author</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Publisher</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Publication Year</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Category</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Total Copies</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Available Copies</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Language</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Description</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Location</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="addButton">
            <property name="styleSheet">
             <string>margin: 10px; padding: 10px; font-size: 14px; background-color: #4CAF50; color: white; border-radius: 5px;</string>
            </property>
            <property name="text">
             <string>Add Book</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deleteButton">
            <property name="styleSheet">
             <string>margin: 10px; padding: 10px; font-size: 14px; background-color: #f44336; color: white; border-radius: 5px;</string>
            </property>
            <property name="text">
             <string>Delete Book</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="UsersTab">
       <attribute name="title">
        <string>Users</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QTableWidget" name="usersTable">
          <property name="editTriggers">
           <set>QAbstractItemView::DoubleClicked</set>
          </property>
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="columnCount">
           <number>10</number>
          </property>
          <column>
           <property name="text">
            <string>User ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>First Name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Last Name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>National ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Email</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Phone</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Address</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Membership Start</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Membership Expiry</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Membership Type</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="addUserButton">
            <property name="styleSheet">
             <string>margin: 10px; padding: 10px; font-size: 14px; background-color: #4CAF50; color: white; border-radius: 5px;</string>
            </property>
            <property name="text">
             <string>Add User</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deleteUserButton">
            <property name="styleSheet">
             <string>margin: 10px; padding: 10px; font-size: 14px; background-color: #f44336; color: white; border-radius: 5px;</string>
            </property>
            <property name="text">
             <string>Delete User</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAdd_From_CSV">
   <property name="text">
    <string>Add From CSV</string>
   </property>
  </action>
  <action name="actionExport_To_CSV">
   <property name="text">
    <string>Export To CSV</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
